crates/runtime/src/task_history.rs

```diff
@@ -169,6 +180,66 @@ impl TaskSpan {
         .boxed()
         .context(UnableToWriteToTableSnafu)?;
 
+        // Override trace_ids if necessary. Must be after above write so that it also handles override this batch of spans.
+        for (from, to) in overrides {
+            Self::override_trace_id(Arc::clone(&df), from, to)
+                .await
+                .boxed()
+                .context(UnableToUpdateTracesSnafu)?;
+        }
+
+        Ok(())
+    }
+
+    async fn override_trace_id(
+        df: Arc<DataFusion>,
+        from: Arc<str>,
+        to: Arc<str>,
+    ) -> Result<(), Error> {
+        let overriden: Vec<_> = df
+            .query_builder(
+                format!(
+                    "SELECT * FROM {} where trace_id = '{from}'",
+                    TableReference::partial(SPICE_RUNTIME_SCHEMA, DEFAULT_TASK_HISTORY_TABLE)
+                        .to_quoted_string()
+                )
+                .as_str(),
+            )
+            .build()
+            .run()
+            .await
+            .boxed()
+            .context(UnableToUpdateTracesSnafu)?
+            .data
+            .try_collect::<Vec<RecordBatch>>()
+            .await
+            .boxed()
+            .context(UnableToUpdateTracesSnafu)?
+            .into_iter()
+            .filter_map(|rb| {
+                // Replace the trace id column with the new `to` trace id.
+                let (idx, _) = rb.schema().column_with_name("trace_id")?;
+                let mut cols = rb.columns().to_vec();
+
+                cols[idx] = Arc::new(StringArray::from(vec![to.to_string(); rb.num_rows()]));
+                Some(RecordBatch::try_new(rb.schema(), cols))
+            })
+            .collect::<Result<_, ArrowError>>()
+            .boxed()
+            .context(UnableToUpdateTracesSnafu)?;
+
+        df.write_data(
+            &TableReference::partial(SPICE_RUNTIME_SCHEMA, DEFAULT_TASK_HISTORY_TABLE),
+            DataUpdate {
+                schema: Arc::new(Self::table_schema()),
+                data: overriden,
+                update_type: UpdateType::Changes,
```

Comment:
lgtm, just a question for learning purposes, is having `span_id` as primary key needed for the changes update here, locating rows with trace_id override changes and make updates?

Reply:
Yes, we don't need it for the initial read (i.e. `WHERE trace_id='{}'`), but is needed so that when we insert these rows back in, we don't get duplicates, but rather the original rows are overwritten.